src/Bridge/Doctrine/Orm/Metadata/Property/DoctrineOrmPropertyNameCollectionFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm\Metadata\Property;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Property\PropertyNameCollection;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+
+/**
+ * Fetch inherited properties in case of a SINGLE_TABLE inheritance
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class DoctrineOrmPropertyNameCollectionFactory implements PropertyNameCollectionFactoryInterface
+{
+    private $reader;
+    private $decorated;
+    private $reflection;
+
+    public function __construct(ManagerRegistry $managerRegistry, PropertyNameCollectionFactoryInterface $decorated = null)
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass, array $options = []) : PropertyNameCollection
+    {
+        if ($this->decorated) {
+            try {
+                $propertyNameCollection = $this->decorated->create($resourceClass, $options);
+            } catch (ResourceClassNotFoundException $resourceClassNotFoundException) {
+                // Ignore not found exceptions from parent
+            }
+        }
+
+        $manager = $this->managerRegistry->getManagerForClass($resourceClass);
+        if (!$manager) {
+            return $propertyNameCollection;
+        }
+
+        $doctrineClassMetadata = $manager->getClassMetadata($resourceClass);
+        if (!$doctrineClassMetadata) {
+            return $propertyNameCollection;
+        }
+
```

Comment:
It can have case where this function return null. I would remove this guard clause and always return `$propertyNameCollection` at the end of the method.